workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs

```diff
@@ -0,0 +1,139 @@
+
+                if (worldCommandResponseHandlers == null)
+                {
+                    continue;
+                }
+
+                foreach (var receivedResponse in reserveEntityIdsResponseData.CommandResponses[i].Responses)
+                {
+                    foreach (var worldCommandHandler in worldCommandResponseHandlers)
+                    {
+                        worldCommandHandler.OnReserveEntityIdsResponseInternal(receivedResponse);
+                    }
+                }
+            }
+
+            for (var i = 0; i < createEntityResponseData.Length; ++i)
+            {
+                var entity = createEntityResponseData.Entities[i];
+                var worldCommandResponseHandlers = GetWorldCommandResponseHandlersForEntity(entity);
+
+                if (worldCommandResponseHandlers == null)
+                {
+                    continue;
+                }
+
+                foreach (var receivedResponse in createEntityResponseData.CommandResponses[i].Responses)
+                {
+                    foreach (var worldCommandHandler in worldCommandResponseHandlers)
+                    {
+                        worldCommandHandler.OnCreateEntityResponseInternal(receivedResponse);
+                    }
+                }
+            }
+
+            for (var i = 0; i < deleteEntityResponseData.Length; ++i)
+            {
+                var entity = deleteEntityResponseData.Entities[i];
+                var worldCommandResponseHandlers = GetWorldCommandResponseHandlersForEntity(entity);
+
+                if (worldCommandResponseHandlers == null)
+                {
+                    continue;
+                }
+
+                foreach (var receivedResponse in deleteEntityResponseData.CommandResponses[i].Responses)
+                {
+                    foreach (var worldCommandHandler in worldCommandResponseHandlers)
+                    {
+                        worldCommandHandler.OnDeleteEntityResponseInternal(receivedResponse);
+                    }
+                }
+            }
+        }
+
+        private WorldCommands.Requirables.WorldCommandResponseHandler[] GetWorldCommandResponseHandlersForEntity(
+            Entity entity)
+        {
+            var entityToReaderWriterStore = gameObjectDispatcherSystem.entityToReaderWriterStore;
+
+            if (!entityToReaderWriterStore.TryGetValue(entity, out var injectableStore))
+            {
+                return null;
+            }
+
+            if (!injectableStore.TryGetInjectablesForComponent(WorldCommandResponseHandlerInjectableId,
+                out var injectables))
+            {
+                return null;
+            }
+
+            if (injectables.Count == 0)
+            {
+                return null;
+            }
+
+            return Array.ConvertAll(injectables.ToArray(),
+                injectable => (WorldCommands.Requirables.WorldCommandResponseHandler) injectable);
```

Comment:
This still iterates over both a hashset and an array and allocates two new arrays in the process. And the result is only used to iterate over the data set once.

Reply:
If we are going to change the backing thing to a list then I don't mind too much what we do here now as it will be changed soon (and world commands aren't sent all that often at the moment)